fiftyone/operators/builtin.py

```diff
@@ -434,90 +1835,428 @@ class LoadWorkspace(foo.Operator):
+                    label=f"This will overwrite existing workspace '{name}'"
+                ),
             )
-        return types.Property(inputs)
+
+        return types.Property(inputs, view=types.View(label="Save workspace"))
 
     def execute(self, ctx):
         name = ctx.params.get("name", None)
         description = ctx.params.get("description", None)
         color = ctx.params.get("color", None)
-        spaces_dict = ctx.params.get("spaces", None)
-        spaces = fo.Space.from_dict(spaces_dict)
-        edit = ctx.params.get("edit", False)
-        current_name = ctx.params.get("current_name", None)
-        if edit:
-            ctx.dataset.update_workspace_info(
-                current_name,
-                info=dict(name=name, color=color, description=description),
-            )
+        spaces = ctx.params.get("spaces", None)
+
+        if isinstance(spaces, dict):
+            spaces = fo.Space.from_dict(spaces)
         else:
-            ctx.dataset.save_workspace(
-                name, spaces, description=description, color=color
-            )
-        return {}
+            spaces = fo.Space.from_json(spaces)
+
+        ctx.dataset.save_workspace(
+            name,
+            spaces,
+            description=description,
+            color=color,
+            overwrite=True,
+        )
 
 
-class DeleteWorkspace(foo.Operator):
+class EditWorkspaceInfo(foo.Operator):
     @property
     def config(self):
         return foo.OperatorConfig(
-            name="delete_workspace", label="Delete Workspace"
+            name="edit_workspace_info",
+            label="Edit workspace info",
+            dynamic=True,
         )
 
     def resolve_input(self, ctx):
         inputs = types.Object()
-        inputs.str("name", label="Workspace Name", required=True)
-        return types.Property(inputs)
+
+        _edit_workspace_info_inputs(ctx, inputs)
+
+        return types.Property(
+            inputs, view=types.View(label="Edit workspace info")
+        )
 
     def execute(self, ctx):
         name = ctx.params.get("name", None)
+        new_name = ctx.params.get("new_name", None)
+        description = ctx.params.get("description", None)
+        color = ctx.params.get("color", None)
+
+        info = dict(name=new_name, description=description, color=color)
+        ctx.dataset.update_workspace_info(name, info)
+
+
+def _edit_workspace_info_inputs(ctx, inputs):
+    workspaces = ctx.dataset.list_workspaces()
+
+    if not workspaces:
+        prop = inputs.str(
+            "msg",
+            label="This dataset has no saved workspaces",
+            view=types.Warning(),
+        )
+        prop.invalid = True
+        return
+
+    workspace_selector = types.AutocompleteView()
+    for key in workspaces:
+        workspace_selector.add_choice(key, label=key)
+
+    # @todo default to current workspace name, if one is currently open
+    inputs.enum(
+        "name",
+        workspace_selector.values(),
```

Comment:
@imanjra @ritch here I'm wanting to add a `default` parameter that specifies the name of the current workspace, if a saved workspace is currently loaded. IE to the equivalent of `default=ctx.view.name` for saved views

Reply:
I believe the current workspace is not available to operator context. This should also be fairly trivial to add.